main.go

```
@@ -56,6 +56,7 @@ var (
 	controllerThreads              = flag.Int("controller-threads", 10, "Number of worker threads used by the SparkApplication controller.")
 	resyncInterval                 = flag.Int("resync-interval", 30, "Informer resync interval in seconds.")
 	namespace                      = flag.String("namespace", apiv1.NamespaceAll, "The Kubernetes namespace to manage. Will manage custom resource objects of the managed CRD types for the whole cluster if unset.")
+	namespaceFilter                = flag.String("namespace-filter", "", "A comma seperated list of regex to manage specific Kubernetes namespaces. Ignored if namespace if specified")
```

Instead of ignoring namespaceFilter when namespace is specified, I would fail fast and throw an error explaining that you shouldn't use both.
If you ignore it, some people might be confused why their filter is not taken into account.

---

"comma seperated list of regex" --> "comma-separated list of regexes"

---

Instead of `""`, you could use the constant `apiv1.NamespaceAll`, like they do in the previous line.

---

-I prefer to ignore/override and avoid failing, apply a regex on a single namespace has no meaning. It has only a meaning if the controller watches/list all namesapces and then filter with respect to the regex.
-Supplying namespace-filter with the constant apiv1.NamespaceAll. Is really miseleading. However, I added some logging, to say if within the filter we have an empty filed: namespace-filter = "" + namespace = apiv1.NamespaceAll => Then we do filter nothing
-Ok for the typo